utils/common/src/main/java/org/apache/brooklyn/util/stream/StreamGobbler.java

```diff
@@ -60,50 +63,84 @@ public void shutdown() {
 
     String logPrefix = "";
     String printPrefix = "";
+
     public StreamGobbler setPrefix(String prefix) {
         setLogPrefix(prefix);
         setPrintPrefix(prefix);
         return this;
     }
+
     public StreamGobbler setPrintPrefix(String prefix) {
         printPrefix = prefix;
         return this;
     }
+
     public StreamGobbler setLogPrefix(String prefix) {
         logPrefix = prefix;
         return this;
-    }    
-    
+    }
+
     @Override
     public void run() {
-        int c = -1;
+        int c, bytes = 0;
+        char[] utfSymbol = new char[2];
         try {
-            while (running.get() && (c=stream.read())>=0) {
-                onChar(c);
+            ByteBuffer bb = ByteBuffer.allocate(4);
+            while (running.get() && (c = stream.read()) >= 0) {
+
+                if (bytes == 0) {
+                    // Identify utf symbol size by Unicode page.
+                    if (c >= 0xF0) {
+                        bytes = 4;
+                    } else if (c >= 0xE0) {
+                        bytes = 3;
+                    } else if (c >= 0xC2) {
+                        bytes = 2;
+                    } else {
+                        bytes = 1;
+                    }
+                }
+
+                //System.out.println(" [" + bb.position() + "] " + Integer.toHexString(c)); // for DEBUG
+                bb.put((byte) c);
+                bytes--;
+
+                if (bytes == 0) {
+                    bb.rewind();
+                    StandardCharsets.UTF_8.decode(bb).get(utfSymbol);
+                    bb.clear();
+                    //System.out.println("[ch] 0x" + Integer.toHexString(utfSymbol[0]) + " 0x" + Integer.toHexString(utfSymbol[1])); // for DEBUG
```

Comment:
@algairim  If this is useful put it in a trace log statement.

Reply:
This code is not needed anymore, removed in ba780b1acac6c42b4d4f5cc8aa0d063339869847.